File: strings/prob1436.py
# 1436 Destination city

class Solution(object):
    def destCity(self, paths):
        """
        :type paths: List[List[str]]
        :rtype: str
        """
        starts = set()
        ends = set()
        for path in paths:
            starts.add(path[0])
            ends.add(path[1])
        diff = ends.difference(starts)
        return list(diff)[0]

File: strings/prob1436.kt
class Solution {
    fun destCity(paths: List<List<String>>): String {
        
        var starts = mutableSetOf<String>()
        var ends = mutableSetOf<String>()
        for ( (start, end) in paths) {
            starts.add(start)
            ends.add(end)
        }
        
        for (end in ends) {
            if (end !in starts) {
                return end
            }
        }
        
        throw Exception("Should not happen")
        
    }
}

class Solution {
    fun destCity(paths: List<List<String>>): String {

        var starts = mutableSetOf<String>()
        var ends = mutableSetOf<String>()
        for ( (start, end) in paths) {
            starts.add(start)
            ends.add(end)
        }

        return (ends subtract starts).toList()[0]
        // throw Exception("Should not happen")

    }
}
